exporters/metric/prometheus/prometheus.go

```diff
@@ -162,8 +163,11 @@ func NewExportPipeline(config Config, period time.Duration) (*push.Controller, h
 	//
 	// Gauges (or LastValues) and Summaries are an exception to this and have different behaviors.
 	integrator := integrator.New(selector, true)
-	pusher := push.New(integrator, exporter, period)
```

Comment:
With a watcher-like API, we should be able to simply replace "period" with a config notifier.